modules/iam/main.tf

```diff
@@ -21,87 +25,58 @@ EOF
   tags               = var.tags
 }
 
+data "aws_iam_policy_document" "sagemaker" {
+  statement {
+    sid = "SagemakerAccess"
+    actions = [
+      "sagemaker:*"
+    ]
+    resources = ["*"]
+  }
 
-resource "aws_iam_policy" "sagemaker_policy" {
-  name   = "${var.model_name}-sagemaker-policy"
-  policy = <<EOF
-{
-    "Version": "2012-10-17",
-    "Statement": [
-	{
-            "Effect": "Allow",
-            "Action": "sagemaker:*",
-            "Resource": "arn:aws:sagemaker:${var.region}:${var.account_id}:*"
-        },
-        {
-            "Effect": "Allow",
-            "Action": "logs:CreateLogGroup",
-            "Resource": "arn:aws:logs:*"
-        },
-        {
-            "Effect": "Allow",
-            "Action": [
-                "logs:CreateLogStream",
-                "logs:PutLogEvents"
-            ],
-            "Resource": [
-                "arn:aws:logs:*"
-            ]
-        },
-	      {
-            "Effect": "Allow",
-            "Action": [
-                "s3:GetObject",
-                "s3:PutObject",
-                "s3:DeleteObject",
-                "s3:ListBucket"
-            ],
-            "Resource": [
-                "arn:aws:s3:::streaming-data-platform-ml-data",
-                "arn:aws:s3:::streaming-data-platform-ml-data/*",
-                "arn:aws:s3:::${var.model_name}-model",
-                "arn:aws:s3:::${var.model_name}-model/*",
-                "arn:aws:s3:::${var.model_name}-config-bucket",
-                "arn:aws:s3:::${var.model_name}-config-bucket/*"
-            ]
-        }, 
-        {
-          "Sid": "AllowPassRole",
-          "Action": "iam:PassRole",
-          "Effect": "Allow",
-          "Resource": "arn:aws:iam::${var.account_id}:role/*",
-          "Condition": {
-            "StringEquals": {
-              "iam:PassedToService": [
-                "sagemaker.amazonaws.com"
```

Comment:
This policy is being attached to a sagemaker instance anyway - I assume this 'passrole' policy was previously being used by a lambda function